hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -521,7 +521,6 @@ protected synchronized byte[] createPassword(TokenIdent identifier) {
    */
```

Comment:
There is a slight chance the currentKey in line 495 and 499 may be inconsistent now that createPassword is no longer synchronized. Can we avoid that by making a local reference to currentKey to ensure within this method we refer to the same object.

Reply:
that's a good catch @jojochuang . Thanks ! It's incorporated.